examples/Madnlp_densekkt_example.jl

```diff
@@ -0,0 +1,87 @@
+using DynamicNLPModels, NLPModels, Random, LinearAlgebra, MadNLP, QuadraticModels, MadNLPGPU, CUDA, SparseArrays
+include("build_thinplate.jl")
+
+
+function MadNLP.jac_dense!(nlp::DenseLQDynamicModel{T, V, M1, M2, M3}, x, jac) where {T, V, M1<: Matrix, M2 <: Matrix, M3 <: Matrix}
+    NLPModels.increment!(nlp, :neval_jac)
+    J = nlp.data.A
+    copyto!(jac, J)
+end
+
+function MadNLP.hess_dense!(nlp::DenseLQDynamicModel{T, V, M1, M2, M3}, x, w1l, hess; obj_weight = 1.0) where {T, V, M1<: Matrix, M2 <: Matrix, M3 <: Matrix}
+    NLPModels.increment!(nlp, :neval_hess)
+    H = nlp.data.H
+    copyto!(hess, H)
+end
+
+function convert_to_CUDA!(lqdm::DenseLQDynamicModel)
+    H = CuArray{Float64}(undef, size(lqdm.data.H))
+    J = CuArray{Float64}(undef, size(lqdm.data.A))
+
+    LinearAlgebra.copyto!(H, lqdm.data.H)
+    LinearAlgebra.copyto!(J, lqdm.data.A)
+
+    lqdm.data.H = H
+    lqdm.data.A = J
+end
+
+# Define attributes of 1 D heat transfer model
+N  = 10
+ns = 5
+nu = 5
+
+dfunc = (i,k)->100*sin(2*pi*(4*i/N-12*k/ns)) + 400
+
+d = zeros(ns, N+1)
+
+for i in 1:(N+1)
+    for j in 1:ns
+        d[j,i] = dfunc(i,j)
+    end
+end
+
+dx = .1
```

Comment:
style guide: prefer writing explicitly `0.1` instead of `.1`. See [here](https://jump.dev/JuMP.jl/dev/developers/style/)